### PR TITLE
Deployment Pipeline

### LINKED ISSUE #118: Automatic Deployment?
Discussion thread for ideas on how automatic deployment (CD) could be set up.

### PR DESCRIPTION
**Pull request by proffapt**
resolves #118 

### Questions

Any path to ignore? Changes there won't trigger this pipeline.

### CODE REVIEW

File: docker-compose.yml
Comment (chirag-ghosh) [inline docker-compose.yml:1]: We don't need to do this change in `docker-compose.prod.yml` file as well?

Comment (proffapt) [reply docker-compose.yml:1]: hmm, yes, made the changes.

**Comment by harshkhandeparkar** [comment]
All paths except the service containers (`/nginx`, `/mediawiki`, `/mysql`, and `/jobs`) and the docker files can be ignored. Currently working on updating the docs and we don't want the wiki to be redeployed each time the readme changes.

**Comment by harshkhandeparkar** [comment]
LGTM otherwise.

**Review by chirag-ghosh — DISMISSED** [review]

**Comment by chirag-ghosh** [comment]
> LGTM otherwise.

Toh bhaiya approve karo na

**Comment by proffapt** [comment]
> All paths except the service containers (`/nginx`, `/mediawiki`, `/mysql`, and `/jobs`) and the docker files can be ignored. Currently working on updating the docs and we don't want the wiki to be redeployed each time the readme changes.

Ok, lemme make this change then

**Comment by proffapt** [comment]
But, docker-compose, docker-compose.prod.yml.. and these things are important too ¯\_(ツ)_/¯ 

So please mention those files which are to be ignored.

**Comment by harshkhandeparkar** [comment]
> But, docker-compose, docker-compose.prod.yml.. and these things are important too ¯_(ツ)_/¯
> 
> So please mention those files which are to be ignored.

Keep the composefiles and ignore the rest.

File: .github/workflows/sync.yaml
Comment (harshkhandeparkar) [inline .github/workflows/sync.yaml:1]: These scripts are not really required on the server. They need to be updated and I am not even sure if they work.

**Review by harshkhandeparkar — APPROVED** [review]

File: .github/workflows/sync.yaml
Comment (proffapt) [reply .github/workflows/sync.yaml:1]: They will be updated someday, and will be required to be present on the server. At that instance of time we will need this sync, I didn't know that these aren't working.. and I have written the workflow which would have been written in that point of time. So, peace.

Comment (harshkhandeparkar) [reply .github/workflows/sync.yaml:1]: I don't think they will be used in the future (at least they won't be run automatically, they are for a user to run manually), but it's fine if they are synced.

**Review by rajivharlalka — APPROVED** [review]

**Pull request merged**